WordPress.com Toolbar: Add Site Logs menu gated behind a constant

p-jackson opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Fixes Automattic/dotcom-forge#1987

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Jetpack plugin can add an optional new Site Logs menu item to the Tools menu.
* Site Logs is only ever added on WoA sites
* Menu item is hidden by default and can be added with the `jetpack_show_wpcom_site_logs_menu` filter

<img width="332" alt="CleanShot 2023-03-22 at 17 57 00@2x" src="https://user-images.githubusercontent.com/1500769/226806156-40bfd11c-5af2-49bf-b2f8-a477f2f80a57.png">

The filter is handled in Automattic/wpcomsh#1315

Having the logic in `wpcomsh` means we can quickly show the menu item to all users without waiting for a Jetpack release.

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->
p9dueE-6YC-p2

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->
No

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

When testing be sure to check the menu both in Calypso and wp-admin.

* [Run these commands on your sandbox](#issuecomment-1478892176) and confirm the Tools menu is unchanged for a sandboxed Simple sites.
* Test on a non-WoA Jetpack site with JN. You can create one with this branch's build already installed. ![CleanShot 2023-03-22 at 17 32 46@2x](https://user-images.githubusercontent.com/1500769/226803416-c6227b05-faa6-424b-811f-0a6f993af5a2.png)
* Test WoA using a dev blog (other docs about testing this on WoA PCYsg-Osp-p2)
  * Install [Jetpack Beta Tester Plugin](https://jetpack.com/download-jetpack-beta/)
  * Configure beta plugin `/wp-admin/admin.php?page=jetpack-beta`
  * Configure Jetpack to use the `add/site-logs-menu` feature branch
  * Deploy Automattic/wpcomsh#1315 to your WoA dev blog
  * If you are on the dev team specified in `wpcomsh`, you should now see the Site Logs menu item in Calypso and wp-admin
  * (after clicking the Site Logs menu item you might be redirected to `/home` unless your testing in a version of Calypso where the page is enabled)

p-jackson committed:
[not verified] Adds new Site Logs menu item on WoA sites

The menu item is gated behind a new `jetpack_show_wpcom_site_logs_menu`
filter.

p-jackson committed:
[not verified] WoA sites opt-in to Site Logs menu by defining a constant

p-jackson committed:
[not verified] changelog

p-jackson committed:
[not verified] Unit test for new filter

p-jackson committed:
[not verified] Add docs for new filter

p-jackson committed:
[not verified] Bump jetpack-mu-wpcom version

zaguiini reviewed (dismissed):
It looks good to me. I'd like to know @jeherve thoughts, though!

p-jackson committed:
[not verified] Remove changes to jetpack-mu-wpcom

It will be more straightforward to deploy the WoA-only code to the wpcomsh plugin directly rather than manage releases.

p-jackson commented:
@zaguiini I've moved the logic for enabling the menu to Automattic/wpcomsh#1315 as suggested p9dueE-6YC-p2#comment-9515. Releasing Site Logs will no longer require all the version bumps this PR used to have.

jeherve commented on projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php:
```diff
@@ -372,6 +372,27 @@ public function add_options_menu() {
+		 * @use add_filter( 'jetpack_show_wpcom_site_logs_menu', '__return_true' );
+		 * @module masterbar
+		 *
+		 * @since 12.0-a.8
```
Let's let the release tool update those numbers for us, this way we're sure it will be the right version, whatever happens.

```suggestion
		 * @since $$next-version$$
```

p-jackson replied:
Thanks! Will the next version be a weekly alpha deployed to Atomic? Or does it come out April 4th?

jeherve reviewed (dismissed):
This tests well for me, although the actual Calypso page doesn't load for me yet. I assume that's still  in development. 

I only have a tiny remark, but I'll commit it myself right and re-approve.

> Test on a non-WoA Jetpack site with JN. You can create one with this branch's build already installed.

Quick tip: you can use [Jetpack live branches](https://github.com/Automattic/jetpack/tree/trunk/tools/jetpack-live-branches) to speed up that process. ;)

jeherve committed:
Update version number in comment
